pkg/api/datasources.go

```diff
@@ -72,6 +72,9 @@ func GetDataSourceById(c *models.ReqContext) response.Response {
 		if errors.Is(err, models.ErrDataSourceNotFound) {
 			return response.Error(404, "Data source not found", nil)
 		}
+		if errors.Is(err, models.ErrDataSourceIdentifierNotSet) {
+			return response.Error(400, "Datasource id is missing.", nil)
```

Comment:
Isn't it better if we standardize to not end HTTP error messages with a period?
```suggestion
			return response.Error(400, "Datasource id is missing", nil)
```